dev/src/util.ts

```diff
@@ -146,3 +146,17 @@ export function isPermanentRpcError(
     return false;
   }
 }
+
+/**
+ * Wraps the provided error in a new error that includes the provided stack.
+ *
+ * Used to preserve stack traces across async calls.
+ * @private
+ */
+export function wrapError(err: GoogleError, stack: string): Error {
+  const wrappedError = new GoogleError(err.message);
+  wrappedError.code = err.code;
+  wrappedError.stack += '\nCaused by: ' + stack;
```

Comment:
Can we modify the stack in 'err' directly?

Reply:
good point. done.